enclave/sgx/eeid_attester.c

```diff
@@ -0,0 +1,178 @@
+
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef OE_BUILD_ENCLAVE
+#include <openenclave/enclave.h>
+#else
+#include <openenclave/host.h>
+#endif
+
+#include <openenclave/attestation/sgx/eeid_attester.h>
+#include <openenclave/bits/eeid.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/sgx/eeid_plugin.h>
+#include <openenclave/internal/sgx/plugin.h>
+#include <openenclave/internal/trace.h>
+
+extern const void* __oe_get_eeid();
+
+#ifdef OE_BUILD_ENCLAVE
+#include <openenclave/attestation/sgx/attester.h>
+#include <openenclave/enclave.h>
+#else
+#include <openenclave/host.h>
+#endif
+
+oe_result_t eeid_attester_on_register(
```

Comment:
Not sure I understand the function name.    Line 43 seems to be the code that does the register, rather than this function being what gets called by the SDK once the plugin is registered.

Reply:
This is the EEID attester plugin. Line 43 registers the SGX attester, which is a requirement for the the EEID attester, so "on register" of EEID we also need to register the SGX attester.